ownCloud/Gallery/AlternativeCollectionCollectionViewController.swift

```diff
@@ -0,0 +1,157 @@
+//
+//  AlternativeCollectionCollectionViewController.swift
+//  ownCloud
+//
+//  Created by Pablo Carrascal on 14/02/2019.
+//  Copyright © 2019 ownCloud GmbH. All rights reserved.
+//
+
+import UIKit
+import ownCloudSDK
+
+private let reuseIdentifier = "Cell"
+
+class AlternativeCollectionCollectionViewController: UICollectionViewController {
+
+	weak private var core: OCCore?
+	private var items: [OCItem]
+	private var query: OCQuery
+	private var index: Int
+
+	// MARK: - Init & deinit
+	init(core: OCCore, items: [OCItem], query: OCQuery, index: Int = 0) {
+		self.core = core
+		self.items = items
+		self.query = query
+		self.index = index
+
+		let layout = UICollectionViewFlowLayout()
+		layout.scrollDirection = .horizontal
+		layout.minimumInteritemSpacing = 0
+		layout.minimumLineSpacing = 0
+		super.init(collectionViewLayout: layout)
+
+		query.addObserver(self, forKeyPath: "hasChangesAvailable", options: [.new], context: nil)
+	}
+
+	deinit {
+		query.removeObserver(self, forKeyPath: "hasChangesAvailable")
+	}
+
+	required init?(coder aDecoder: NSCoder) {
+		fatalError("init(coder:) has not been implemented")
+	}
+
+	// MARK: - ViewController LifeCycle
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        // Register cell classes
+        self.collectionView!.register(GalleryCollectionViewCell.self, forCellWithReuseIdentifier: reuseIdentifier)
+		self.collectionView.isPagingEnabled = true
+		self.collectionView.scrollToItem(at: IndexPath(row: index, section: 0), at: .centeredHorizontally, animated: false)
+    }
+
+    // MARK: UICollectionViewDataSource
+    override func numberOfSections(in collectionView: UICollectionView) -> Int {
+        return 1
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return items.count
+    }
+
+    override func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath) as? GalleryCollectionViewCell
+
+		let itemToDisplay = items[indexPath.row]
+		let mimeType = itemToDisplay.mimeType!
+
+		let viewController = self.selectDisplayViewControllerBasedOn(mimeType: mimeType)
+		let shouldDownload = viewController is (DisplayViewController & DisplayExtension) ? true : false
+
+		var configuration: DisplayViewConfiguration
+		if !shouldDownload {
+			configuration = DisplayViewConfiguration(item: itemToDisplay, core: core, state: .notSupportedMimeType)
+		} else {
+			configuration = DisplayViewConfiguration(item: itemToDisplay, core: core, state: .hasNetworkConnection)
+		}
+
+		if cell?.item == nil {
+			cell?.item = itemToDisplay
```

Comment:
@pablocarmu not sure how it looks like but I am thinking, that may be it is a bit of overkill resource-wise if we have complex view hierarchies added to collection view cells, suppose it is multiple PDF documents and we have multiple PDFViewController views? Or is my thinking wrong?

Reply:
Yes, this is why I finally choose the `UIPageViewController` way, what I don't know Is why this collection view approach is still here, maybe I wrong pushed it, sorry. Will address the other comments you wrote and push it with this removed.

Reply:
Seems that Xcode removed only the reference of these files and this is what this collection view approach was uploaded. It should be out of the code now.